mem-cache: Add unit test for LRURP

odanrc opened the pull request:
As side effects:
- We now verify that the repl data is not null

odanrc committed:
mem-cache: Add unit test for LRURP

As side effects:
- We now verify that the repl data is not null

Signed-off-by: Daniel R. Carvalho <odanrc@yahoo.com.br>

erin-le reviewed (commented):
@odanrc: Hello, the tests for LRU RP generally look good to me. I tried running these tests locally, and it looks like the build fails due to the dynamic_pointer_cast on line 54 of lru_rp.test.cc. Could you take a look? Could you also remove the changes for LFU RP from this PR? Thank you!

odanrc committed:
mem-cache: Remove LRU repl data validation

Since the base class is not polymorphic, and we don't want it
to be, we cannot rely on dynamic_cast to validate the type of
the entry created. As such, we are forced to remove this
validation.

Signed-off-by: Daniel R. Carvalho <odanrc@yahoo.com.br>

odanrc commented:
> @odanrc: Hello, the tests for LRU RP generally look good to me. I tried running these tests locally, and it looks like the build fails due to the dynamic_pointer_cast on line 54 of lru_rp.test.cc. Could you take a look? Could you also remove the changes for LFU RP from this PR? Thank you!

I wish I could remove the changes for LFU, but I am completely unable to change the base branch to the unit_test_lfu_rp branch, so the changes from the LFU PR, which I want to merge first, ends up showing in this PR too. Otherwise, I have removed the dynamic cast, like in the LFU PR.

erin-le commented on src/mem/cache/replacement_policies/lru_rp.test.cc:
```diff
@@ -0,0 +1,310 @@
+#include "params/LRURP.hh"
+
+// We need to initialize the event queue to be able to count ticks
+gem5::EventQueue eventQueue("MRURPTest Queue");
```
Should this be:

```suggestion
gem5::EventQueue eventQueue("LRURPTest Queue");
```

odanrc replied:
Now you know where I got my template from :)

Fixed

erin-le reviewed (dismissed):
LGTM

odanrc committed:
mem-cache: Fixup event Q name in LRU test

There was a typo.

Signed-off-by: Daniel R. Carvalho <odanrc@yahoo.com.br>
